### PR TITLE
AppScan 360 support

### PR DESCRIPTION
**Pull request by vishalhcl-5960**

### CODE REVIEW

File: src/main/java/com/hcl/appscan/sdk/auth/IAuthenticationProvider.java [hunk 1]
```diff
@@ -45,4 +45,6 @@ public interface IAuthenticationProvider {
 	 * @return The proxy
 	 */
 	public Proxy getProxy();
+
+    public boolean getCertificates();
```
Comment (mattmurp) [inline src/main/java/com/hcl/appscan/sdk/auth/IAuthenticationProvider.java:49]: Can you provide information about what this method does? It's not clear from the name and there is no javadoc comment on it.

File: src/main/java/com/hcl/appscan/sdk/scanners/sast/SAClient.java [hunk 2]
```diff
@@ -104,7 +104,17 @@ private int runClient(String workingDir, List<String> args, String irGenClient,
 			m_builder.environment().put(IRGEN_CLIENT_PLUGIN_VERSION, irgenClientPluginVersion);
 			
 		m_progress.setStatus(new Message(Message.INFO, Messages.getMessage(PREPARING_IRX, getLocalClientVersion())));
-		final Process proc = m_builder.start();
+        if(properties.get("certificates")!=null && properties.get("certificates").equals("true")){
```
Comment (mattmurp) [inline src/main/java/com/hcl/appscan/sdk/scanners/sast/SAClient.java:107]: Similar to above, we seem to be doing the exact same thing in both the if and the else case, with the only difference being the addition of the "-Dacceptssl" option. Can you consolidate this code so any checks we do simply append that option and we're not duplicating code in both the if and else cases?

File: src/main/java/com/hcl/appscan/sdk/auth/IAuthenticationProvider.java (hunk 1 above)
Comment (mattmurp) [inline src/main/java/com/hcl/appscan/sdk/auth/IAuthenticationProvider.java:49]: Also, changing this interface means that every existing implementation will need to implement this method, so is it something that's absolutely needed?

Comment (vishalhcl-5960) [reply src/main/java/com/hcl/appscan/sdk/auth/IAuthenticationProvider.java:49]: Matt, Based on this method, we are appending the "-Dacceptssl" in the server URL which has been used to set APPSCAN_OPTS in the SA_Client class.
Note: I have modified the name of this method for better understanding.

Comment (vishalhcl-5960) [reply src/main/java/com/hcl/appscan/sdk/auth/IAuthenticationProvider.java:49]: Yeah Matt, It is required. For this, i had also added this method in the ASEAuthenticationProvider class which always return "False".

File: src/main/java/com/hcl/appscan/sdk/scanners/sast/SAClient.java (hunk 2 above)
Comment (vishalhcl-5960) [reply src/main/java/com/hcl/appscan/sdk/scanners/sast/SAClient.java:107]: Agree, Matt. I have consolidated the code accordingly.

**Review by kripajoym — APPROVED** [review]

**Pull request merged**